### ISSUE TITLE
[exporter/awsemf] Exporter ignores the first batch of a metric send to cloudwatch

### ISSUE THREAD

**Issue by mircohaug** [aws-observability/aws-otel-collector#1653]
The `awsemf` exporter of the ADOT distribution of OTEL sends faulty data for the first batch of metrics send to AWS Cloudwatch.

## Steps to reproduce
0. The code is checked in at https://github.com/mircohaug/otel-awsemf-reproduction-ignored-first-batch
1. Authenticate your shell against an AWS account with the `default` profile. (Or change the value of the `AWS_PROFILE` env var in step 3)
2. create the log group `emfbug-reproduction-embedded-metrics-otel` by running `aws logs create-log-group --log-group-name emfbug-reproduction-embedded-metrics-otel`
3. Start the otel agent by running  
```
docker run -d --rm -p 4317:4317 \
-e AWS_REGION=eu-central-1 \
-e AWS_PROFILE=default \
-v ~/.aws:/root/.aws \
-v "$(pwd)/otel-agent-config.yaml":/otel-local-config.yaml \
--name awscollector \
public.ecr.aws/aws-observability/aws-otel-collector:latest \
--config otel-local-config.yaml;
```
4. run `yarn install`
5. create metrics by running `yarn start`
   1. In [the code](./src/index.ts) we create a new OTEL counter. We add `1` to the counter four times in total. We Split these four increments on two batches with two increments each. Between these batches there is a wait time to allow the OTEL agent to flush the values to AWS.
6. Run this [log insights query](https://eu-central-1.console.aws.amazon.com/cloudwatch/home?region=eu-central-1#logsV2:logs-insights$3FqueryDetail$3D$257E$2528end$257E0$257Estart$257E-1800$257EtimeType$257E$2527RELATIVE$257Eunit$257E$2527seconds$257EeditorString$257E$2527fields*20counter_name*2c*40timestamp$257EisLiveTail$257Efalse$257EqueryId$257E$2527293e02d9-1cfc-446f-bb07-174daa3d4833$257Esource$257E$2528$257E$2527emfbug-reproduction-embedded-metrics-otel$2529$2529) (`fields counter_name,@timestamp` on the log group `emfbug-reproduction-embedded-metrics-otel` ) to see the published EMF Metrics.
   1. See actual and expected result 
7. Cleanup by running `docker rm -f awscollector` and `aws logs delete-log-group --log-group-name emfbug-reproduction-embedded-metrics-otel` 

## Expected Result
We expect the loggroup to contain two entries with a value of `2`. One entry for each batch.

## Actual result
We only get one entry with a value of `2`. The first batch gets its value set to `0`. 

### Additional information
* The behaviour persists over multiple runs of the script. 
* Also does it happen for each new combination of counter name, and attributes. 
* In Order to exclude a faulty implementation in the OTEL Framework we also added a `ConsoleMetricExporter` alongside the one that exports the metrics to the OTEL agent. This Exporter print the correct values to the console.
* In addition we added a `file` exporter to the OTEL agent pipeline. This one also shows the correct values.
* Small waits between the batches lead to all four increments ending up in the same batch and we se no value in cloudwatch whatsoever.
* The [upstream issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16218). Opened here as well for visibility.

**Comment by bryan-aguilar** [comment]
Could this be due to the fact that the  awsemfexporter performs [cululative-to-delta conversion](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awsemfexporter/datapoint.go#L126-L129) for [cumulative sums](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awsemfexporter/datapoint.go#L238-L240)? You can see more relevant discussion here on this [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3425).

**Comment by mircohaug** [comment]
This sounds exactly like the behavior I've observed. Do you @bryan-aguilar  know if this can be disabled or if there is a workaround? I already tried to to initialize the counter and adding 0 before using it but this did not work. Probably because the values are aggregated over a certain time frame...

**Comment by jan-xyz** [comment]
I'm running into the same bug as well. I was actually not seeing any metrics at all until I found your bug report and tried to just collect some more data points. 

I tried force flushing once when starting up the application as a workaround but it also doesn't "fix" it.

**Comment by mircohaug** [comment]
PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17988 was merged. In the next release there exists the configuration flag `retain_initial_value_of_delta_metric` in the emf subsection. If this flag is set to true the initial value of any metric is not dropped any more.

**Comment by mizzzto** [comment]
Hi @mircohaug 
I tried your test project by adding the new flag  retain_initial_value_of_delta_metric: true in [otel-agent-config.yaml](https://github.com/mircohaug/otel-awsemf-reproduction-ignored-first-batch/blob/main/otel-agent-config.yaml)
I added the flag in the awsemf section right below namespace. However, after running the application the CloudWatch log was still showing incorrect count. Have you tested the application after the PR was merged and released? Or am I missing some other configuration needed?
